src/clj_kondo/impl/linters.clj

```diff
@@ -354,6 +354,20 @@
                   (and call?
                        (not (utils/linter-disabled? call :single-logical-operand))
                        (lint-single-logical-operand call))]]
+      (when (and (not (utils/linter-disabled? ctx :two-argument-reduce))
+                 (= fn-name 'reduce)
+                 (= 2 arity)
+                 (#{'clojure.core 'cljs.core} fn-ns))
```

Comment:
Although this looks nice, I try to avoid this kind of set construct because of performance reasons, it's much faster to just do the boring thing here and writing it fully with `or` and `=`.

I think it makes sense to do this check in `analyzer`, we're already "casing" on core things there.

Reply:
That's why i put it in last position in the `and`, but i'm also fine with the boring variant, np.

When first looking into it, I tried to put this into `analyze-hof` right above the arity checks, which didn't work... now i see i misunderstood the meaning of `arity` in that context... moving to analyzer.